dnf5/commands/advisory_shared.hpp

```diff
@@ -66,6 +69,16 @@ inline std::optional<libdnf5::advisory::AdvisoryQuery> advisory_query_from_cli_i
         if (!advisory_names.empty()) {
             auto advisories_names = libdnf5::advisory::AdvisoryQuery(base);
             advisories_names.filter_name(advisory_names);
+            if (advisories_names.empty()) {
+                const BgettextMessage msg = M_("No advisory found matching the requested name: \"{}\"");
```

Comment:
I like that this gives a concrete message.
I think it would be good to do something similar for `advisory_bzs` and `advisory_cves`.

In addition would it make sense to check if the returned advisory query is not empty (if there is one)?
Unless I am missing something if the query is empty no packages will get past that filter. I also think it can make the other message a little bit confusing: `No match for argument 'pkg' in advisories` but all advisories are filtered out.

Reply:
Yes, adding such a warning for both bzs and cves makes sense in the current state.

However, if we eventually decide to change the behavior to use intersection between different advisory filters instead of the current union (as discussed in https://github.com/rpm-software-management/dnf5/issues/1401), this could become a bit confusing. For example, if an advisory that fixes a given BZXY is filtered out due to a severity mismatch, the message "No advisory matching BZXY" might be misleading.

But I agree that providing some form of feedback when no advisory matches the given filters would be helpful to users.

Reply:
> However, if we eventually decide to change the behavior to use intersection between different advisory filters instead of the current union (as discussed in https://github.com/rpm-software-management/dnf5/issues/1401), this could become a bit confusing.

Since I believe we are sticking with the current compatible version do you plan to add the mentioned warnings for bzs and cves or are we fine without them? I wouldn't consider it a blocker.

However I did notice the CI tests update now needs an update after the messages changed.

Reply:
I will add the warnings for BZs and CVEs.
With regard to https://bugzilla.redhat.com/show_bug.cgi?id=2362585, does it make sense for `advisory list/info UNKNOWN-ADVISORY` commands to also return exit code 1?

Reply:
> Or maybe more generally - return exit code 1 in case an advisory filter was specified, but the returning query is empty

That is a good question. Looking at https://github.com/rpm-software-management/dnf5/issues/1075 if all the query commands always return zero maybe we should keep it consistent?

Reply:
Thanks for the pointer! Let me summarize what should happen when the filters result in an empty query across the different locations where advisory filters are used:

1. `advisory XXX` commands – return code 0
2. `check-upgrade` command – return code 0
3. `repoquery` command – return code 0
4. `install` command – similar to installing an unknown package – return code 1
5. `upgrade` command – similar to upgrading an unknown package – return code 1

In each case, a specific warning should be printed to stderr to inform the user about non-existent name/BZ/CVE.

Reply:
That sound very good to me. :+1: 

(I am also wondering if `check-upgrade` shouldn't behave more like `upgrade` with regards to exit codes, if `check-upgrade unknown-package` should rather return code 1, but that is a separate thing.)

Reply:
I'm still struggling with the `upgrade` use-case. There are situations (especially when no package spec is given), when having the empty advisories query is completely OK - e.g. `dnf upgrade --security`. So I guess the upgrade command with an empty advisories query is more like upgrade with no upgrades available in repos, then upgrade of an unknown package. Maybe this even apply to the `install` command from the original #784 issue.

Currently I think that we should not touch the exit codes and just print a warning/info message that the user entered non-existent advisory or not fixed BZ/CVE.

I should have think about it more before filing this PR...

Reply:
I though the exit codes would change only when the advisory name/BZ/CVE doesn't match anything. It seemed to me that matches the unknown package case well.
So the advisory query can still be empty (which is valid as you said, but informing the user is nice) however when a specific advisory is requested and its not there it is an error.

Due to the queries being ORed it can also happen that the query is not empty but there was a wrong advisory name, like: `dnf upgrade --security --advisory=wrong-name`. To me this seems analogous to an upgrade of multiple packages where one has a typo: `dnf upgrade htop zsh unknown-package`. This is by default an error (controlled by `--skip-unavailable`).

Although maybe I am missing something its getting quite complex.

Reply:
Maybe I'm just overthinking it. Added BZ/CVE warning message.